src/others/observable/lv_observable.h

```diff
@@ -0,0 +1,263 @@
+
+/**
+ * Initialize a color type subject
+ * @param subject address of the subject
+ * @param color initial value
+ */
+void lv_subject_init_color(lv_subject_t* subject, lv_color_t color);
+
+/**
+ * Subscribe an LVGL widget to a subject
+ * @param subject address of the subject
+ * @param cb callback to call when the value of the subject changes
+ * @param obj the LVGL object what will be affected by the change
+ * @param data1 optional additional parameter stored in the observer object
+ * @param data2 optional additional parameter stored in the observer object
+ * @return the observer object
+ * @note This is a generic interface, which is normally not called directly. For specific purposes there are specific wrappers.
+ * @note The observer object is allocated on the heap. It will be freed automatically when the object is deleted or when it is unsubscibed from the subject.
+ */
+void* lv_subject_subscribe_obj(lv_subject_t* subject, lv_observer_cb_t cb, lv_obj_t* obj, void* data1, void* data2);
+
+/**
+ * Remove observer from the list of subscribers
+ * @param subject address of the observer object returned when a widget was subscribed to the subject
+ */
+void lv_observer_unsubscribe(lv_observer_t* subject);
+
+/**
+ * Remove all observers of a specified widget from the list of observers
+ * @param subject address of the subject
+ * @param obj the LVGL widget
+ */
+void lv_subject_unsubscribe_obj(lv_subject_t* subject, lv_obj_t* obj);
+
+/**
+ * Remove all observers of a subject
+ * @param subject address of the subject
+ */
+void lv_subject_unsubscribe_all(lv_subject_t* subject);
+
+/**
+ * Update the value of an int type subject
+ * @param subject address of the subject
+ * @param value new value
+ * @note automatically notifies all observers
+ */
+void lv_subject_set_int(lv_subject_t* subject, int32_t value);
+
+/**
+ * Update the value of a pointer type subject
+ * @param subject address of the subject
+ * @param value new value
+ * @note automatically notifies all observers
+ */
+void lv_subject_set_ptr(lv_subject_t* subject, void* value);
+
+/**
+ * Update the value of a color type subject
+ * @param subject address of the subject
+ * @param color new value
+ * @note automatically notifies all observers
+ */
+void lv_subject_set_color(lv_subject_t* subject, lv_color_t color);
+
+/**
+ * Copy the string into the buffer of a string type subject
+ * @param subject address of the subject
+ * @param buf string to copy 
+ * @note automatically notifies all observers
+ */
+void lv_subject_copy_string(lv_subject_t* subject, char* buf);
+
+/**
+ * Set bitmask and value comparison pattern of the observer
+ * @param observer address of the observer
+ * @param pattern pattern used to mask out interested bits from the bitmask's value and compare it with a constant value
+ * @note The value
+ */
+void lv_observer_bitmask_set_cond(lv_observer_t* observer, lv_bitmask_pattern_t pattern);
```

Comment:
It seems it's not used anymore.

Reply:
There was a mismatch between the header and the implementation. Now I have renamed it to 'lv_observer_bitmask_set_pattern'.